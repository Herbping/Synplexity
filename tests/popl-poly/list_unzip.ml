type 'a list =
  | Nil
  | Cons of 'a * 'a list

let rec list_unzip : ('a * 'b) list -> ('a list * 'b list) |>
  /\([] -> ([], []),
     [(a, b)] -> ([a], [b]),
     [(c, d); (a, b)] -> ([c; a], [d; b])) = ?
