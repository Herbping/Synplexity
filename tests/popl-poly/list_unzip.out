(* Generated Thu Oct  8 20:27:06 2015 *)
fix f1 (l1:'a * 'b list) : 'a list * 'b list ->
  (match l1 with
  | Nil u1 -> Nil
  | Cons t1 -> Cons (#1 (#1 t1), #1 (f1 (#2 t1))), match l1 with
  | Nil u1 -> Nil
  | Cons t1 -> Cons (#2 (#1 t1), #2 (f1 (#2 t1))))
